src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -620,6 +620,12 @@ private static ImageHelper createImageProvider(JlinkConfiguration config,
                 String msg = taskHelper.getMessage("err.runtime.link.jdk.jlink.prohibited");
                 throw new IllegalArgumentException(msg);
             }
+            // Do not permit linking from run-time image when the current image
+            // is being patched
+            if (jdk.internal.misc.VM.getSavedProperties().get("jdk.module.patch.0") != null) {
```

Comment:
Yes, I'm wondering if a JDK-specific (rather than standard) property could be more widely useful as it would allow observability tools to know that the JDK has been patched. It would be good to get wider input on this.

Reply:
https://mail.openjdk.org/pipermail/core-libs-dev/2024-November/134623.html

Reply:
@AlanBateman The discussion doesn't see to gain much traction. Should I go ahead and propose a bug/patch - with CSR -  which adds the JDK specific property and make it dependent on this fix? Thoughts?

Reply:
I've now proposed a property in [JDK-8344560](https://bugs.openjdk.org/browse/JDK-8344560) and based this patch on it.